src/Storefront/Resources/views/components/Sw/Product/BuyButton.js

```diff
@@ -0,0 +1,44 @@
+class BuyButton extends ShopwareComponent {
+
+    static selector = 'form[data-component="BuyButton"]';
+
+    static options = {
+        redirectSelector: '[name="redirectTo"]',
+        redirectParamSelector: '[data-redirect-parameters="true"]',
+        redirectTo: 'frontend.cart.offcanvas',
+    };
+
+    init() {
+        this.redirectInput = this.el.querySelector(this.options.redirectSelector);
+        this.redirectParamInput = this.el.querySelector(this.options.redirectParamSelector);
+
+        if (this.redirectInput) {
+            this.redirectInput.value = this.options.redirectTo;
+        }
+
+        if (this.redirectParamInput) {
+            this.redirectParamInput.disabled = true;
+        }
+
+        this.el.addEventListener('submit', this.onFormSubmit.bind(this));
+    }
+
+    destroy() {
+        this.el.removeEventListener('submit', this.onFormSubmit.bind(this));
+    }
+
+    onFormSubmit(event) {
+        event.preventDefault();
+
+        let requestUrl = this.el.getAttribute('action');
+        let formData = window.Shopware.serializeForm(this.el);
+
+        [ requestUrl, formData ] = window.Shopware.emitInterception('BuyButton:PreSubmit', requestUrl, formData);
+
+        window.Shopware.emit('BuyButton:Submit', requestUrl, formData);
+
+        window.Shopware.callPluginMethod('OffCanvasCart', 'openOffCanvas', requestUrl, formData);
```

Comment:
Not sure about mixing the 2 systems directly like this. This is needed here because otherwise you are in this endless chain of re-building everything with the new system like the OffCanvasCart opening mechanism. 

But I would find it better to directly use the `PluginManager` if we really need to call a method. This way it looks like something of the new component system in reality it is using the PluginSystem. And we introduce yet another API to call the old API from the new API.